searchsummary/src/vespa/searchsummary/docsummary/rankfeaturesdfw.cpp

```diff
@@ -2,36 +2,30 @@
 
 #include "rankfeaturesdfw.h"
 #include "docsumstate.h"
-#include <vespa/searchlib/common/packets.h>
 #include <vespa/vespalib/data/slime/cursor.h>
 
 namespace search::docsummary {
 
-RankFeaturesDFW::RankFeaturesDFW() :
-    _env(nullptr)
+RankFeaturesDFW::RankFeaturesDFW(IDocsumEnvironment * env) :
+    _env(env)
 { }
 
 RankFeaturesDFW::~RankFeaturesDFW() = default;
 
-void
-RankFeaturesDFW::init(IDocsumEnvironment * env)
-{
-    _env = env;
-}
-
 void
 RankFeaturesDFW::insertField(uint32_t docid, GetDocsumsState *state,
                              ResType type, vespalib::slime::Inserter &target)
 {
-    if (state->_rankFeatures.get() == nullptr) {
+    if ( !state->_rankFeatures ) {
         state->_callback.FillRankFeatures(state, _env);
         if (state->_rankFeatures.get() == nullptr) { // still no rank features to write
             return;
         }
     }
     const FeatureSet::StringVector & names = state->_rankFeatures->getNames();
     const FeatureSet::Value * values = state->_rankFeatures->getFeaturesByDocId(docid);
-    if (type == RES_FEATUREDATA && values != nullptr) {
+    assert(type == RES_FEATUREDATA);
```

Comment:
i would not assert here as this is configured, not computed. same with other assert.